### ISSUE TITLE
Update catch2 version to accommodate newer glibc version

### ISSUE THREAD

**Issue by jandrej** [mfem/mfem#3104]
See https://github.com/catchorg/Catch2/issues/2178

**Comment by pazner** [comment]
We have the fix mentioned in the linked issue. Is there a different problem happening?

**Comment by jandrej** [comment]
@samreynoldsmath reported that with Ubuntu 22 glib he's seeing this error 
```g++ -g -c /home/sam/Libraries/mfem-4.4/tests/unit/unit_test_main.cpp  -O3 -std=c++11 -I../..  -I. -I../.. -o unit_test_main.o
In file included from /usr/include/signal.h:328,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/catch.hpp:8030,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/unit_tests.hpp:15,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/run_unit_tests.hpp:15,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/unit_test_main.cpp:14:
/home/sam/Libraries/mfem-4.4/tests/unit/catch.hpp:10818:58: error: call to non-‘constexpr’ function ‘long int sysconf(int)’
10818 |     static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
      |                                                          ^~~~~~~~~~~
In file included from /usr/include/x86_64-linux-gnu/bits/sigstksz.h:24,
                 from /usr/include/signal.h:328,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/catch.hpp:8030,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/unit_tests.hpp:15,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/run_unit_tests.hpp:15,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/unit_test_main.cpp:14:
/usr/include/unistd.h:640:17: note: ‘long int sysconf(int)’ declared here
  640 | extern long int sysconf (int __name) __THROW;
      |                 ^~~~~~~
In file included from /home/sam/Libraries/mfem-4.4/tests/unit/unit_tests.hpp:15,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/run_unit_tests.hpp:15,
                 from /home/sam/Libraries/mfem-4.4/tests/unit/unit_test_main.cpp:14:
/home/sam/Libraries/mfem-4.4/tests/unit/catch.hpp:10877:45: error: size of array ‘altStackMem’ is not an integral constant-expression
10877 |     char FatalConditionHandler::altStackMem[sigStackSize] = {};
      |                                             ^~~~~~~~~~~~
make[2]: *** [makefile:134: unit_test_main.o] Error 1
make[2]: Leaving directory '/home/sam/Libraries/mfem-4.4/tests/unit'
make[1]: *** [makefile:446: tests/unit] Error 2
make[1]: Leaving directory '/home/sam/Libraries/mfem-4.4'
make: *** [makefile:515: test] Error 2
```

**Comment by samreynoldsmath** [comment]
I used the "Latest Release" .tgz file downloaded directly from mfem.org. Following @pazner 's suggestion, I have instead cloned the master branch and reconfigured (the serial version). All tests passed!
